### PR TITLE
🌱 Replace github.com/pkg/errors

### PR DESCRIPTION
**Pull request by pierreprinetti**
**What this PR does / why we need it**:

The package [pkg/errors](https://github.com/pkg/errors) has served its purpose and its proposed changes have been incorporated into Go's standard library in Go v1.13.

This change removes it as a direct dependency, replacing its functionality with the corresponding features of `fmt` and `errors`.

**Special notes for your reviewer**:

The old `pkg/errors` defined the method `Cause() error`, whereas the stdlib incorporated its functionality with the method `Unwrap() error`. As such, if a consumer of CAPO directly called `errors.Cause(err)` to match the causing error, then their code would silently break because the underlying sentinel error wouldn't be identified.

This problem is susceptible to affect any code that uses CAPO as a library, that uses  a version of `pkg/errors` below [`v0.9.0`](https://github.com/pkg/errors/releases/tag/v0.9.0).

The last versions of `pkg/errors` (v0.9.0+) [incorporated](https://github.com/pkg/errors/blob/614d223910a179a466c1767a985424175c39b465/errors.go#L248) a compatibility layer with the new `Unwrap()` syntax.

**TODOs**:

- [x] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

_EDIT: Added a note pointing out that there is a risk of compatibility issues_

### CODE REVIEW

**Comment by pierreprinetti** [comment]
Incidentally, with this change the CI is now checking the style of the errors... which is why I had to remove the capital letter from an error message :)

File: controllers/openstackcluster_controller.go [hunk 1]
```diff
@@ -232,24 +232,24 @@ func deleteBastion(scope scope.Scope, cluster *clusterv1.Cluster, openStackClust
 		for _, address := range addresses {
 			if address.Type == corev1.NodeExternalIP {
 				if err = networkingService.DeleteFloatingIP(openStackCluster, address.Address); err != nil {
-					handleUpdateOSCError(openStackCluster, errors.Errorf("failed to delete floating IP: %v", err))
-					return errors.Errorf("failed to delete floating IP: %v", err)
+					handleUpdateOSCError(openStackCluster, fmt.Errorf("failed to delete floating IP: %w", err))
+					return fmt.Errorf("failed to delete floating IP: %v", err)
```
Comment (lentzi90) [inline controllers/openstackcluster_controller.go:236]:
```suggestion
					return fmt.Errorf("failed to delete floating IP: %w", err)
```

**Review by lentzi90 — COMMENTED** [review]
Looks like a quite small change for us so I'm in favor of taking this in. From what I understand CAPI and kubeadm would have much larger changes so they are waiting for now. But I see plenty of PRs like this in k/k so I think it makes sense for us to follow.

File: controllers/openstackcluster_controller.go [hunk 2]
```diff
@@ -443,16 +443,16 @@ func reconcileNetworkComponents(scope scope.Scope, cluster *clusterv1.Cluster, o
 		netOpts := openStackCluster.Spec.Network.ToListOpt()
 		networkList, err := networkingService.GetNetworksByFilter(&netOpts)
 		if err != nil {
-			handleUpdateOSCError(openStackCluster, errors.Errorf("failed to find network: %v", err))
-			return errors.Errorf("failed to find network: %v", err)
+			handleUpdateOSCError(openStackCluster, fmt.Errorf("failed to find network: %w", err))
+			return fmt.Errorf("failed to find network: %w", err)
 		}
 		if len(networkList) == 0 {
-			handleUpdateOSCError(openStackCluster, errors.Errorf("failed to find any network: %v", err))
```
Comment (pierreprinetti) [inline controllers/openstackcluster_controller.go:450]: wait. `err` is `nil` here, isn't it? Why did we wrap it, here and below?

File: controllers/openstackcluster_controller.go (hunk 1 above)
Comment (pierreprinetti) [reply controllers/openstackcluster_controller.go]: hawk eye!

File: controllers/openstackcluster_controller.go (hunk 2 above)
Comment (pierreprinetti) [reply controllers/openstackcluster_controller.go:450]: I have changed this error (and its message) in a separate commit. PTAL!

**Comment by jichenjc** [comment]
/approve

Agree it's small change and we can follow other k/k behavior..
leave the unhold to other folks in case they want to take another look

**Review by lentzi90 — COMMENTED** [review]
/hold cancel

**Pull request merged**